obspy/io/gcf/core.py

```diff
@@ -33,68 +142,584 @@ def merge_gcf_stream(st):
+         # release allocated memory
+         gcf_io.free_GcfFile(obj)
+   return is_gcf
 
-def _read_gcf(filename, headonly=False, **kwargs):  # @UnusedVariable
-    """
-    Reads a GCF file and returns a Stream object.
 
-    only GCF files containing data records are supported.
-
-    .. warning::
+def _read_gcf(filename, networkcode='', stationcode='', locationcode='', bandcode="H", instrumentcode="H", channel_prefix=None, 
+              blockmerge=True, headonly=False, cleanoverlap=True, errorret=False, **kwargs):
+   """
+   Reads a GCF file and returns a :class:`~obspy.core.stream.Stream`
+   
+   Only GCF data records are supported. Function supports format as described by 
+   GCF Reference `SWA-RFC-GCFR Issue F, December 2021 <https://www.guralp.com/apps/ok?doc=GCF_format>`
+      
+   .. warning::
         This function should NOT be called directly, it registers via the
         ObsPy :func:`~obspy.core.stream.read` function, call this instead.
-
-    :type filename: str
-    :param filename: GCF file to be read.
-    :type headonly: bool, optional
-    :param headonly: If True read only head of GCF file.
-    :type channel_prefix: str, optional
-    :param channel_prefix: Channel band and instrument codes.
-        Defaults to ``HH``.
-    :rtype: :class:`~obspy.core.stream.Stream`
-    :returns: Stream object containing header and data.
-
-    .. rubric:: Example
+   
+   :type filename: str
+   :param filename: path to GCF file to read
+   :type networkcode: str, optional
+   :param networkcode: network code to use 
+   :type stationcode: str, optional
+   :param stationcode: station code to use, if not specified unitID in gcf header will be used (first 4
+   :   characters in streamID)
+   :type locationcode: str, optional
+   :param locationcode: location code to use
+   :type bandcode: str, optional
+   :param bandcode: 1-character band code to use, ignored if channel_prefix is input
+   :type instrumentcode: str, optional
+   :param instrumentcode: 1-character instrument code to use, ignored if channel_prefix is input
+   :type channel_prefix: str, optional
+   :param channel_prefix: 2-character channel prefix to use, if not input channel_prefix = bandcode+instrumentcode
+   :type blockmerge: bool, optional
+   :param blockmerge: if True merge blocks if aligned
+   :type headonly: bool, optional
+   :param headonly: if True only read block headers
+   :type cleanoverlap: bool, optional
+   :param cleanoverlap: ignored if blockmerge is False, if True, remove overlapping identical data prior to merging blocks, 
+       if False overlapping blocks will not be merged even if data in overlap is identical
+   :type errorret: bool, optional
+   :param errorret: if True block and header issues will be set in trace.stats.gcf.stat for each :class:`~obspy.core.trace.Trace`
+       object in returned :class:`~obspy.core.stream.Stream` object, else function will raise raise an IOError with an appropriate
+       error message
```

Comment:
in some readers we have `strict` for things like that I believe.. maybe we could use that wording? or maybe rename to `raise_on_error`?

Reply:
Sorry I do not really follow, can you elaborate?